dask_geopandas/geohash.py

```diff
@@ -0,0 +1,221 @@
+    coords = np.array([y_mids, x_mids]).T
+    # Encode coords with Geohash
+    geohash = encode_geohash(coords, precision)
+
+    return pd.Series(geohash, index=gdf.index, name="geohash")
+
+
+def encode_geohash(coords, precision):
+    """
+    Calculate geohash based on coordinates for a
+    given precision
+
+    Parameters
+    ----------
+    coords : array_like
+    precision : int
+        precision of the Geohash
+
+    Returns
+    ---------
+    geohash: array containing geohashes for each coordinate
+    """
+
+    quantized_coords = _encode_quantize_points(coords)
+    g_uint64 = _encode_into_uint64(quantized_coords)
+    gs_uint8_mat = _encode_base32(g_uint64)
+    geohash = _encode_unicode(gs_uint8_mat, precision)
+
+    return geohash
+
+
+def _encode_quantize_points(coords):
+    """
+    Quantize coordinates by mapping
+    unit intervals [0, 1] & multiplying by 2^32.
+
+    Parameters
+    ----------
+    coords : array_like
+        coordinate pairs
+
+    Returns
+    ---------
+    quantized_coords : array_like
+        quantized coordinate pairs
+    """
+
+    _q = np.array([(2.0 ** 32 / 180, 0), (0, 2.0 ** 32 / (180 * 2))], dtype="float64")
+
+    quantized_coords = coords + np.array((90, 180))
+    quantized_coords = np.dot(quantized_coords, _q)
+    quantized_coords = np.floor(quantized_coords)
+
+    return quantized_coords
+
+
+def _encode_into_uint64(quantized_coords):
+    """
+    Encode quantized coordinates into uint64
+
+    Implementation based on "Geohash in Golang Assembly"
+    blog (https://mmcloughlin.com/posts/geohash-assembly)
+
+    Parameters
+    ----------
+    quantized_mat : array_like
+        coordinate pairs
+
+    Returns
+    ---------
+    quantized_coords : array_like
+        quantized coordinate pairs
+    """
+
+    __s1 = np.array([(1, 0), (0, 2)], dtype="uint64")
+
+    g_uint64 = np.uint64(quantized_coords)
+    g_uint64 = g_uint64.reshape(-1, 2)
+    g_uint64 = np.bitwise_and(
+        np.bitwise_or(g_uint64, np.left_shift(g_uint64, 16)), 0x0000FFFF0000FFFF
+    )
+    g_uint64 = np.bitwise_and(
+        np.bitwise_or(g_uint64, np.left_shift(g_uint64, 8)), 0x00FF00FF00FF00FF
+    )
+    g_uint64 = np.bitwise_and(
+        np.bitwise_or(g_uint64, np.left_shift(g_uint64, 4)), 0x0F0F0F0F0F0F0F0F
+    )
+    g_uint64 = np.bitwise_and(
+        np.bitwise_or(g_uint64, np.left_shift(g_uint64, 2)), 0x3333333333333333
+    )
+    g_uint64 = np.bitwise_and(
+        np.bitwise_or(g_uint64, np.left_shift(g_uint64, 1)), 0x5555555555555555
+    )
+    g_uint64 = np.dot(g_uint64, __s1)
+    g_uint64 = np.bitwise_or(g_uint64[:, 0], g_uint64[:, 1])
+    g_uint64 = np.right_shift(g_uint64, 4)
+
+    return g_uint64
+
+
+def _encode_base32(g_uint64):
+    """
+    Encode quantized coordinates into uint64
+
+    Implementation based on "Geohash in Golang Assembly"
+    blog (https://mmcloughlin.com/posts/geohash-assembly)
+    Parameters
+    ----------
+    quantized_mat : array_like
```

Comment:
Here and elsewhere, please make sure that your parameters in the docstring match the parameter names declared above for the function.  Also please consider more human readable parameter names, I'm finding `g_uint64` a bit inscrutable here.

Reply:
updated in fca27ecb0590e625f15d7db255b9dcb11312fd74